Sources/XcbeautifyLib/Renderers/GitHubActionsRenderer.swift

```diff
@@ -0,0 +1,279 @@
+import Foundation
+
+struct GitHubActionsRenderer: OutputRendering {
+    private enum AnnotationType: String {
+        case notice
+        case warning
+        case error
+    }
+
+    let colored: Bool = false
+
+    private func outputGitHubActionsLog(
+        annotationType: AnnotationType,
+        fileComponents: FileComponents? = nil,
+        message: String
+    ) -> String {
+        let formattedFileComponents = fileComponents?.formatted ?? ""
+        return "::\(annotationType) \(formattedFileComponents)::\(message)"
+    }
+
+    func formatCompileError(group: CompileErrorCaptureGroup, additionalLines: @escaping () -> (String?)) -> String {
+        let filePath = group.filePath
+        let fileComponents = filePath.asFileComponents()
+        let reason = group.reason
+
+        // Read 2 additional lines to get the error line and cursor position
+        let line: String = additionalLines() ?? ""
+        let cursor: String = additionalLines() ?? ""
```

Comment:
You definitely have a better understanding of this parser than I do at this point, so I'm gonna ask a couple questions to help me better understand:

- Is reading `additionalLines` into a variable a common pattern throughout the codebase already?
- What happens if the format of these lines change within Xcode's output? Are there ways to safeguard against potential future changes?

Reply:
> Is reading additionalLines into a variable a common pattern throughout the codebase already?

Yes it is. That's how the pre-existing implementation (now `TerminalRenderer`) is implemented.
 https://github.com/tuist/xcbeautify/blob/a49d6e54189da4f208e61e98bc23c11fd6ab9d0d/Sources/XcbeautifyLib/Renderers/TerminalRenderer.swift#L241-L242

> What happens if the format of these lines change within Xcode's output? Are there ways to safeguard against potential future changes?

There aren't any. We'd need to add additional logic, but I think that's out of the scope of this PR. I'm not sure it's even something worth implementing unless the Xcode output actually changes.